timelock-agent/src/main/java/com/palantir/timelock/config/SqlitePaxosPersistenceConfiguration.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.config;
+
+import java.io.File;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@Value.Immutable
+// TODO (jkong): Allow serialization/deserialization, once we want this to actually be configurable
+public interface SqlitePaxosPersistenceConfiguration {
+    /**
+     * Indicates where the SQLite paxos state log files should be stored.
+     * Should not conflict with any other data directories that may be used by TimeLock to store Paxos state logs.
+     */
+    @JsonProperty("data-directory")
+    File dataDirectory();
```

Comment:
Is there any other config that you need here? Would it not make sense to inline this?

Reply:
I'm not aware of anything right now, but I also don't know that I won't need anything more

Reply:
I would be surprised if we wanted more config, the only thing I can think of is tuning any sqlite options, but instead we should just hardcode them and make sure they work correctly.

Reply:
Going ahead with this for now, we can change later on if needed: I'm not confident there is a single correct set of parameters given variance in request loads.